sticker-utils/src/save.rs

```diff
@@ -0,0 +1,90 @@
+use failure::Fallible;
+
+use sticker::tensorflow::TaggerTrainer;
+
+#[derive(Copy, Clone, Eq, PartialEq)]
+pub enum CompletedUnit {
+    /// A batch is completed.
+    Batch,
+
+    /// An epoch is completed.
+    Epoch,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq)]
+pub enum SaveSchedule {
+    /// Save after every N batches.
+    Batches(usize),
+
+    /// Save after every epoch.
+    Epoch,
+
+    /// Save every epoch and after every N batches.
+    EpochAndBatches(usize),
+}
+
+/// Scheduler that saves at points dictated by the schedule.
+pub struct SaveScheduler {
+    epoch_batch: usize,
+    epoch: usize,
+    batch: usize,
+    schedule: SaveSchedule,
+}
+
+impl SaveScheduler {
+    /// Current batch.
+    pub fn batch(&self) -> usize {
+        self.batch
+    }
+
+    /// Current epoch.
+    pub fn epoch(&self) -> usize {
+        self.epoch
+    }
+
+    /// Save the model parameters when a save point has been reached.
+    pub fn save(&mut self, trainer: &TaggerTrainer, completed: CompletedUnit) -> Fallible<()> {
+        match completed {
+            CompletedUnit::Epoch => {
+                match self.schedule {
+                    SaveSchedule::Epoch | SaveSchedule::EpochAndBatches(_) => {
+                        trainer.save(format!("pretrain-epoch-{}", self.epoch))?
```

Comment:
Should the prefix `pretrain` be an argument? As far as I can tell, this is also the prefix for saved epochs when using `sticker-train`.

Reply:
Good catch! will fix that tomorrow!

Reply:
Fixed.